packages/webcert/src/feature/certificate/Inputs/UeCauseOfDeath/UeCauseOfDeathControl.tsx

```diff
@@ -88,7 +95,8 @@ const UeCauseOfDeathControl: React.FC<Props> = ({
   }
 
   const handleSpecificationChange = (code: string) => {
-    onChange({ ...value, specification: { ...value.specification, code } })
+    const specificationId = config.specifications.find((s) => s.code === code)?.id ?? ''
+    onChange({ ...value, specification: { ...value.specification, id: specificationId, code: code } })
```

Comment:
I just noticed that we are setting `item.id` as value in each option element, so what we are calling `code` here is really "id" right now. Perhaps we can just set an index to the options value and select specification from the config.

```tsx
const handleSpecificationChange = ({ id, code }: ConfigUeCodeItem) => {
  onChange({ ...value, specification: { ...value.specification, id, code } })
}

<Dropdown 
onChange={(event) => {
  handleSpecificationChange(specifications[parseInt(event.currentTarget.value)])
}}
options={specifications.map((item, index) => (
  <option value={index} key={item.id}>
    {item.label}
  </option>
))}
/>
```

Reply:
This won't work, since we set the value of the dropdown to specification.code, so mapping the option value to index won't match
![image](https://user-images.githubusercontent.com/92843575/205319382-a727871b-416a-4d47-b097-8367e2ac8108.png)

Reply:
Good point! :)